### PR TITLE
Add IMDS named tags & ExtendedMeta

### LINKED ISSUE #954: No way to fetch values of named Tags in IMDS Instance Metadata
Hi!

Just got to trying the 2.0.0 release, well done :muscle: Sincerely appreciated.

An immediate issue with EC2 Metadata: using amazonka, it's **impossible to fetch actual values** of instance `Tags`.

To showcase what I mean, here I'm primarily interested in the `Name` instance tag:
```
$ curl http://instance-data/latest/meta-data/tags/instance/Name
app-dev-04
# ↑↑↑ the Name tag value I need to fetch in app

$ curl http://instance-data/latest/meta-data/tags/instance/
Cost_center
Customer
Environment
Name
Type
# ↑↑↑ all instance tags, \n-separated... keys only! no values!
```

Given this:

https://github.com/brendanhay/amazonka/blob/e081929d5a81f30397c9c9a219b154da1a922fb8/lib/amazonka/src/Amazonka/EC2/Metadata.hs#L576-L587

— it's only possible to do the 2nd curl, obtaining the result `"Cost_center\nCustomer\nEnvironment\nName\nType"` — which for my purpose is (obviously, I hope) utterly useless. The 1st curl which I need, is impossible to express.

I've attempted to work around by:
```haskell
data InstanceNameTag = InstanceNameTag
instance AmazonkaCore.ToText InstanceNameTag where toText _ = "instance/Name"

    {- ... -}
    instanceName <- IMDS.metadata man InstanceNameTag
```
— but ofcourse, it's a type error :confused: 
```
src/Application/Metrics.hs:159:37: error:
    • Couldn't match expected type ‘Metadata’
                  with actual type ‘InstanceNameTag’
    • In the second argument of ‘metadata’, namely ‘InstanceNameTag’
      In a stmt of a 'do' block:
        instanceName <- metadata man InstanceNameTag
      In the expression:
        do instanceId <- metadata man InstanceId <&> decodeUtf8
           machineName <- metadata man (Tags Instance) <&> showT
           instanceName <- metadata man InstanceNameTag
           let (internalIpv4, internalPort) = ...
           ....
    |       
159 |   instanceName <- IMDS.metadata man InstanceNameTag
    |                                     ^^^^^^^^^^^^^^^
```

--------------------------------------------------------------------------

Would it sound sensible to add a constructor like so
```haskell
data Tags
  = Instance
  | InstanceTag !Text
```
— to allow querying any named tag using amazonka? Or extend `Metadata` instead?

Or am I missing some other way to extend here on the side of library-user? 

I can develop a patch and file a PR if any of above sounds any good. Just let me know.

### PR DESCRIPTION
**Pull request by ulidtko**
Hi @endgame — submitting a little patch that fixes #954.

Both discussed items implemented. Compiles & tests pass. Please review :pray:

### CODE REVIEW

File: lib/amazonka/src/Amazonka/EC2/Metadata.hs [hunk 1]
```diff
@@ -214,6 +214,12 @@ data Metadata
     Spot !Spot
   | -- | See: 'Tags'
     Tags !Tags
+  | -- | Any other piece of metadata specified by arbitrary key.
+    --
+    -- This is provided for forward compatibility: should there ever appear any gap
+    -- between a future version of AWS API and what (possibly an outdated version of)
+    -- this library covers -- the constructor gives the option to work around.
+    XtendedMeta !Text
```
Comment (endgame) [inline lib/amazonka/src/Amazonka/EC2/Metadata.hs:222]:
```suggestion
    -- This is provided for forward compatibility: should there ever appear any gap
    -- between a future version of the IMDS API and what this library covers,
    -- this constructor provides a workaround.
    Other !Text
```

**Review by endgame — COMMENTED** [review]
Please also add an entry to `lib/amazonka/CHANGELOG.md`.

Note: I'll be away and uncontactable for a couple of weeks from tomorrow.

**Comment by ulidtko** [comment]
@endgame done.

File: lib/amazonka/CHANGELOG.md [hunk 2]
```diff
@@ -14,6 +14,7 @@
   Before this fix you had to create an empty credentials file for these methods to work correctly.
 - `amazonka`: During logging, do not print bytestrings that have unprintable characters
 [\#952](https://github.com/brendanhay/amazonka/pull/952)
+- `amazonka`: Support reading values of IMDS metadata tags, add `Metadata.Other` constructor [\#955](https://github.com/brendanhay/amazonka/pull/955)
```
Comment (endgame) [inline lib/amazonka/CHANGELOG.md:17]:
```suggestion
- `amazonka`: Support reading values of IMDS metadata tags, add `Metadata.Other` constructor
[\#955](https://github.com/brendanhay/amazonka/pull/955)
```

**Review by endgame — APPROVED** [review]

**Comment by ulidtko** [comment]
Thanks!

**Pull request merged**